### ISSUE TITLE
Calling maybeAuth and friends in version 1.6

### ISSUE THREAD

**Issue by paul-rouse** [yesodweb/yesod#1488]
Am I being totally thick (entirely possible :smile:)?  What is the intended way to upgrade this sort of code to version 1.6, now that `maybeAuth` is in `AuthHandler`?

```
getHomeR :: Handler Html
getHomeR = do
    mauth <- maybeAuth
    ...
```

I can write a function that runs it, but it messes with the `RunHandlerEnv`, which hardly seems right!  Have I missed something?

**Comment by snoyberg** [comment]
Great catch, this is definitely a bug in the API. I'll push out a fix shortly I hope.

**Comment by paul-rouse** [comment]
I did wonder whether the function I wrote could be made general, but I think it would have to be generated somewhere in the template haskell, since, at least the way I've done it, it bakes in the master route constructor, and therefore the foundation type.  Probably not ideal, though.

```
runAuthHandler :: SubHandlerFor Auth App a -> HandlerFor App a
runAuthHandler (SubHandlerFor f) = HandlerFor $ \hd -> f hd
      { handlerEnv =
          let rhe = handlerEnv hd
          in rhe { rheRoute = case rheRoute rhe of
                                Just (AuthR subR) -> Just subR
                                _                 -> Nothing
                 , rheRouteToMaster = AuthR
                 , rheChild = getAuth (rheSite rhe)
                 }
      }

getHomeR :: Handler Html
getHomeR = do
    mauth <- runAuthHandler maybeAuth
```

**Comment by snoyberg** [comment]
I've pushed a commit which I believe addresses this, a3f130233bbf4e6bfebdb7bfd3d926c689b53fd7. Can you give it a shot?

**Comment by paul-rouse** [comment]
Great, yes, that works perfectly in the example I was trying (the [test](https://github.com/paul-rouse/yesod-auth-hashdb/blob/master/test/TestSite.hs) for `yesod-auth-hashdb`).

Thanks!

**Comment by snoyberg** [comment]
Awesome, thanks for confirming. I've just released version 1.6.1.
